File: tests/patterns/swift/equivalence_constant_propagation_class_fields.swift
class A {
  private var f1: String
  private let f3 = "abc"
  private var f4: String
  var f5: String
  private let f7: String
  private var f8 = "abc"
  private static var f2: String = "abc"
  init() { f1 = "abc"; f4 = "abc"; f5 = "abc"; f7 = "abc"; self.f9 = "abc" }
  private var f9: String
  func other() { f4 = "zzz" }
  func m() {
    // ERROR: match
    let loc = "abc"; sink(0, loc)
    sink(1, f1)
    sink(2, A.f2)
    // ERROR: match
    sink(3, f3)
    sink(4, f4)
    sink(5, f5)
    // ERROR: match
    sink(7, f7)
    sink(8, f8)
    sink(9, self.f9)
  }
}
extension A {
  func ext() { sink(10, f1); sink(11, self.f3) }
}

File: tests/patterns/dart/equivalence_constant_propagation_class_fields.dart
class A {
  String _f1;
  final String _f3 = "abc";
  String _f4;
  String f5;
  final String _f7;
  String _f8 = "abc";
  static String _f2 = "abc";
  A() { _f1 = "abc"; _f4 = "abc"; f5 = "abc"; }
  A.named() : _f7 = "abc" { _f1 = "zzz"; }
  void other() { _f4 = "zzz"; }
  void m() {
    // ERROR: match
    var loc = "abc"; sink(0, loc);
    sink(1, _f1);
    sink(2, _f2);
    // ERROR: match
    sink(3, _f3);
    sink(4, _f4);
    sink(5, f5);
    sink(8, _f8);
  }
}
class B {
  String _g1;
  B() { _g1 = "abc"; }
  void m() { sink(10, _g1); }
}
